### ISSUE TITLE
romio fails to build due to missing stdatomic.h

### ISSUE THREAD

**Issue by naughtont3** [open-mpi/ompi#10373]
when building an ompi-5.0.x snapshot i get errors about building romio due to missing `stdatomic.h`.  

This seems like a new dependency/build require. Should this have been caught during configure phase?

### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)

 - openmpi-5.0.0 nightly tarball 12-may-2022

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

 - nightly tarball
 - gcc/g++ 4.8.5 (i know it is old)

### Please describe the system on which you are running

* Operating system/version: Linux
* Computer hardware: x86-64
* Network type: 

-----------------------------

## Details of the problem

A build of openmpi fails during compile phase due to missing `stdatomic.h` while building 3rd-party/romio341.

```shell
./configure \
       --enable-debug \
       --with-devel-headers \
    --enable-mpirun-prefix-by-default \
    --prefix=${OMPI_INSTALL_DIR} \

 ...<snip>...

make[2]: Entering directory `/home/3t4/projects/ompi-x/scratch/CLOUDY-FALCON/openmpi-v5.0.x-202205120246-ef29ef7/build2/3rd-party/romio341'
Making all in mpl
make[3]: Entering directory `/home/3t4/projects/ompi-x/scratch/CLOUDY-FALCON/openmpi-v5.0.x-202205120246-ef29ef7/build2/3rd-party/romio341/mpl'
  CC       src/atomic/mpl_atomic.lo
  CC       src/bt/mpl_bt.lo
  CC       src/env/mpl_env.lo
  CC       src/dbg/mpl_dbg.lo
In file included from ../../../../3rd-party/romio341/mpl/include/mpl_atomic.h:101:0,
                 from ../../../../3rd-party/romio341/mpl/src/atomic/mpl_atomic.c:11:
../../../../3rd-party/romio341/mpl/include/mpl_atomic_c11.h:20:23: fatal error:stdatomic.h: No such file or directory
 #include <stdatomic.h>
                       ^
compilation terminated.
In file included from ../../../../3rd-party/romio341/mpl/include/mpl_atomic.h:101:0,
                 from ../../../../3rd-party/romio341/mpl/include/mpl.h:15,
                 from ../../../../3rd-party/romio341/mpl/src/env/mpl_env.c:6:
../../../../3rd-party/romio341/mpl/include/mpl_atomic_c11.h:20:23: fatal error:stdatomic.h: No such file or directory
 #include <stdatomic.h>
                       ^
compilation terminated.
In file included from ../../../../3rd-party/romio341/mpl/include/mpl_atomic.h:101:0,
                 from ../../../../3rd-party/romio341/mpl/include/mpl.h:15,
                 from ../../../../3rd-party/romio341/mpl/src/dbg/mpl_dbg.c:12:
../../../../3rd-party/romio341/mpl/include/mpl_atomic_c11.h:20:23: fatal error:stdatomic.h: No such file or directory
 #include <stdatomic.h>
                       ^
compilation terminated.
In file In file In file included from ../../../../3rd-party/romio341/mpl/include/mpl_atomic.h:101:0,
                 from ../../../../3rd-party/romio341/mpl/include/mpl.h:15,
                 from ../../../../3rd-party/romio341/mpl/src/bt/mpl_bt.c:6:
../../../../3rd-party/romio341/mpl/include/mpl_atomic_c11.h:20:23: fatal error:stdatomic.h: No such file or directory
 #include <stdatomic.h>
                       ^
compilation terminated.
make[3]: *** [src/atomic/mpl_atomic.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: *** [src/env/mpl_env.lo] Error 1
make[3]: *** [src/dbg/mpl_dbg.lo] Error 1
make[3]: *** [src/bt/mpl_bt.lo] Error 1
make[3]: Leaving directory `/home/3t4/projects/ompi-x/scratch/CLOUDY-FALCON/openmpi-v5.0.x-202205120246-ef29ef7/build2/3rd-party/romio341/mpl'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/home/3t4/projects/ompi-x/scratch/CLOUDY-FALCON/openmpi-v5.0.x-202205120246-ef29ef7/build2/3rd-party/romio341'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/3t4/projects/ompi-x/scratch/CLOUDY-FALCON/openmpi-v5.0.x-202205120246-ef29ef7/build2/3rd-party'
make: *** [all-recursive] Error 1
```

**Comment by jsquyres** [comment]
@naughtont3 Perfectly valid.  configure should elect not to build ROMIO in this case.  Sounds like a legit bug.

**Comment by awlauria** [comment]
FWIW - it looks like it is a bug specific to gcc v4.8, not present on gcc v4.9.

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=58016
https://gcc.gnu.org/gcc-4.9/changes.html

**Comment by ggouaillardet** [comment]
And FWIW, from 3rd-party/romio341/mpl/include/mpl_atomic.h

```
#if defined(MPL_USE_NO_ATOMIC_PRIMITIVES)
#include "mpl_atomic_none.h"
#elif defined(MPL_HAVE_C11_ATOMICS)
#include "mpl_atomic_c11.h"
#elif defined(MPL_HAVE_GCC_INTRINSIC_ATOMIC)
#include "mpl_atomic_gcc_atomic.h"
```

That seems to point at a `configure` but when detecting C11 atomics

**Comment by ggouaillardet** [comment]
The root cause is `3rd-party/romio341/mpl/include/mplconfig.h` should not be in the tarball (this is an automatically generated file)

That being said, I am not sure why this file is not re-generated at `configure` time, and if it is and `VPATH` is used, why the file in the source directory is used instead of the one in the build directory.

For the time being, a workaround is to remove this file after untarring Open MPI and before running `configure`

**Comment by awlauria** [comment]
@naughtont3 this got overlooked, merged here into v5.0.x: https://github.com/open-mpi/ompi/pull/11323

if it turns out the fix come from upstream ompio like we would prefer, we'll revert it out. closing.
